### DISCUSSION TITLE
Проблема з «великими» ролями

### DISCUSSION THREAD

**Discussion by capitoshko** [IRPTeam/Roles#5] [Ideas]
Якщо у ролі більше 99999 значень прав доступу, тоді виникає посилка при записуванні в табличну частину Rights довідник ролей. Вирішив проблему додаванням додаткових 2-х табличних частин і протягуванням їх по кількох модулях. Можливо комусь буде корисним у роботі з нетиповими та само дописаними конфігураціями типу УВП.

**Comment by DitriXNew** [comment]
Якщо справді таке є, то це проблема з ролями, треба їх вивчити та зрозуміти – чому там так багато всього. І найімовірніше – почистити їх. У такому разі це може навіть прискорити роботу системи.
Тому я б спочатку дослідив цю проблему.

**Reply by capitoshko** [reply]
Ми розуміємо, що з ролями велика біда. І ця помилка - це не помилка розробника, це саме те сміття, що ми назбирали протягом років. Але щоб її дослідити з допомогою вашого механізму, то спочатку ролі потрібно завантажити і провести подальше дослідження.

**Reply by DitriXNew** [reply]
Для цього достатньо вивантажити конфігурацію у XML. Потім знайти необхідні файли ролей. Написати найпростіший скрипт, який видалить усі місця, де є false. Чи навіть зробити це саморуч.

**Comment by capitoshko** [comment]
Погоджуюся, це також гарний варіант вирішення проблеми.
